wire-library/wire-runtime-swift/src/main/swift/ProtoReader.swift

```diff
@@ -745,7 +778,66 @@ public final class ProtoReader {
             throw ProtoDecoder.Error.mapEntryWithoutKey(value: value)
         }
         guard let unwrappedValue = value else {
-            throw ProtoDecoder.Error.mapEntryWithoutValue(key: key)
+            throw ProtoDecoder.Error.mapEntryWithoutValue(key: unwrappedKey)
+        }
+
+        // Preallocate a few empty slots to avoid reallocations.
+        if dictionary.isEmpty && MemoryLayout<V>.size <= ProtoReader.collectionPreallocationMaxValueSize {
+            dictionary.reserveCapacity(ProtoReader.collectionPreallocationSlotCount)
+        }
+
+        dictionary[unwrappedKey] = unwrappedValue
+    }
+
+    /// Decode a key-value pair where the value is RawReprentable with RawValue == Int32.
+    /// - Parameters:
+    ///   - dictionary: The dictionary in which to add key-value pair if the decoded value is a known case in V.
+    ///   - decodeKey: A closure that decodes the key for each pair
+    ///   - addUnknownPair: A closure that adds to the key-value pair to unknown fields in the event a given raw value is not a known case in V.
```

Comment:
typo: "adds to the key-value pair" -> "adds the key-value pair". I'd also probably clarify to something like "in the event a given value represents an unknown enum case and the enum decoding strategy is set to `returnsNil`."